src/components/PaymentDropdown.tsx

```diff
@@ -0,0 +1,178 @@
+'use client';
+import { useState, useEffect } from 'react';
+import { Button } from '@/components/ui/button';
+import { Trash, ChevronDown, ChevronUp } from 'lucide-react';
+import { deleteSolanaAddress, deleteUpiId } from '@/actions/payoutMethods';
+import { SolanaAddress, UpiId } from '@prisma/client';
+import { useAction } from '@/hooks/useAction';
+import { toast } from 'sonner';
+import NewPayoutDialog from '@/components/NewPayoutDialog';
+
+interface PaymentMethodsDropdownProps {
+  upiAddresses: UpiId[] | undefined;
+  solanaAddresses: SolanaAddress[] | undefined;
+  fetchPayoutMethods: () => void;
+}
+
+export default function PaymentMethodsDropdown({
+  upiAddresses,
+  solanaAddresses,
+  fetchPayoutMethods,
+}: PaymentMethodsDropdownProps) {
+  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
+  const [isPayoutDialogOpen, setIsPayoutDialogOpen] = useState<boolean>(false);
+  const [btnClicked, setBtnClicked] = useState<string>('');
+
+  const { execute: executeDeleteUPI } = useAction(deleteUpiId, {
+    onSuccess: () => {
+      toast.success('UPI Address deleted successfully');
+      fetchPayoutMethods();
+    },
+    onError: () => {
+      toast.error('Failed to delete UPI address');
+    },
+  });
+
+  const { execute: executeDeleteSolana } = useAction(deleteSolanaAddress, {
+    onSuccess: () => {
+      toast.success('Solana Address deleted successfully');
+      fetchPayoutMethods();
+    },
+    onError: () => {
+      toast.error('Failed to delete Solana address');
+    },
+  });
+
+  const handleUpiDelete = (id: number) => {
+    executeDeleteUPI({ id });
+  };
+
+  const handleSolanaDelete = (id: number) => {
+    executeDeleteSolana({ id });
+  };
+
+  useEffect(() => {
```

Comment:
This effect seems unnecessary.

Reply:
first i had the payment dialog open by default so this was required, now the only relevant updates to payout methods happen as a result of dialog interactions so this is irrelevant. tested it by removing this useeffect will push the changes now